kornia/core/image.py

```diff
@@ -0,0 +1,338 @@
+        return self.data.shape[-3]
+
+    @property
+    def height(self) -> int:
+        """Return the image height (columns)."""
+        return self.data.shape[-2]
+
+    @property
+    def resolution(self) -> Tuple[int, int]:
+        """Return a tuple with the image (height, width)."""
+        return (self.height, self.width)
+
+    @property
+    def width(self) -> int:
+        """Return the image width (rows)."""
+        return self.data.shape[-1]
+
+    @property
+    def color(self) -> ImageColor:
+        """Return the color space representation of the image."""
+        return self._meta['color']
+
+    @color.setter
+    def color(self, x: ImageColor) -> None:
+        """Setter for the color space representation of the image."""
+        self._meta['color'] = x
+
+    # TODO: figure out a better way map this function
+    def float(self) -> 'Image':
+        return Image(self.data.float(), self.color, self._mean(), self._std())
+
+    def _mean(self) -> List[float]:
+        return self._meta['mean']
+
+    def _std(self) -> List[float]:
+        return self._meta['std']
+
+    @classmethod
+    def from_numpy(
+        cls, data, color: ImageColor, mean: Optional[List[float]] = None, std: Optional[List[float]] = None
+    ) -> 'Image':
+        """Construct an image tensor from a numpy array.
+
+        Args:
+            data: a numpy array with the shape of :math:`(H,W,C).
+            color: the color space representation of the input data.
+            mean: a list containing the mean values of the input data.
+            std: a list containing the standard deviation of the input data.
+
+        Example:
+            >>> import numpy as np
+            >>> data = np.ones((4, 5, 3), dtype=np.uint8)  # HxWxC
+            >>> img = Image.from_numpy(data, color=ImageColor.BGR)
+            >>> assert img.channels == 3
+            >>> assert img.resolution == (4, 5)
+            >>> assert img.shape == (3, 4, 5)
+        """
+        return cls(image_to_tensor(data), color, mean, std)
+
+    def to_numpy(self):
+        """Return a numpy array with the shape of :math:`(H,W,C)`."""
+        return tensor_to_image(self.data, keepdim=True)
+
+    @classmethod
+    def from_dlpack(
+        cls, data, color: ImageColor, mean: Optional[List[float]] = None, std: Optional[List[float]] = None
+    ):
+        """Construct an image tensor from a DLPack capsule.
+
+        Args:
+            data: a DLPack capsule from numpy, tvm or jax.
+            color: the color space representation of the input data.
+            mean: a list containing the mean values of the input data.
+            std: a list containing the standard deviation of the input data.
+
+        Example:
+            >>> import numpy as np
+            >>> x = np.ones((4, 5, 3))
+            >>> img = Image.from_dlpack(x.__dlpack__(), ImageColor.RGB)
+        """
+        from torch.utils.dlpack import from_dlpack
```

Comment:
why import it here?

Reply:
i'll move on the top -- we need to design well this api to be flexible in front of different frameworks taking ownership of the tensor data. Also what would be the expected input shape or if we want to be flexible to support both CxHxW and HxWxC

Reply:
as first step I'll simplify this PR and we improve slowly slowly based on more community feedback. Images apis are very opinionated too depending on the area you work :)

Reply:
I don't remember the name, but I had seen something which should help with this question about channels -- but simplifying it would be something like having a list or string with the name of each channel

but for now, I would start simple and keep the kornia pattern